### DISCUSSION TITLE
Common practice for maintainer

### DISCUSSION THREAD

**Discussion by ryansurf** [ryansurf/cli-surf#37] [Q&A] [answered]
So I've been working on code and pushing it quite often. I do sometimes break the app when I oversee some bug (like [here](https://github.com/ryansurf/cli-surf/discussions/25#discussioncomment-9615531)), should I also be submitting PRs for review? Not really sure what the common practice is on this one, what do you guys think?

@K-dash @GUKWAT

**Comment by K-dash** [comment, marked as answer]
@ryansurf 
Yes, that's right. Without a review process, there's a risk of missing bugs and breaking the app. Therefore, even if you're the project owner, it's a common practice to submit PRs and merge them into the main branch only after they've been reviewed and approved by a reviewer (maintainer).

While that's fine for personal development, it can cause issues for external contributors if a bug occurs in an OSS project...😱

If possible, could you add me as an "external collaborator" (maintainer) for this repository? That way, I can be assigned as a reviewer using GitHub's functionality, and I would be happy to take on the review tasks.

**Reply by ryansurf** [reply]
Yes! Will set that up now

**Reply by ryansurf** [reply]
Invitation sent :smiley:

Just wondering, do you work as a dev? You seem pretty informed with all the new tech/best practices, I'm still learning a lot of common practices

**Reply by K-dash** [reply]
Thank you for inviting me! I hope we can work together to make cli-surf an even more interesting service 🥳

> Just wondering, do you work as a dev? You seem pretty informed with all the new tech/best practices, I'm still learning a lot of common practices

Yes, I work as an IT engineer in Japan. I have about 6 years of experience as an infrastructure engineer and 4 years as a software engineer (mainly backend).
Actually, my work is not very busy until June, so I think I can cooperate with you in various ways (lol).

As a heads-up, I may become a bit busy from July onward, which could delay my pull request reviews. If it's urgent, please feel free to skip the review and merge. 😶

**Reply by ryansurf** [reply]
That would be awesome!

Nice, that makes a lot of sense. I graduated a few months ago, pretty new to the open source and collaboration world. 

I was just in Japan not too long ago! Only for a few days, I visited Tokyo and Kyoto.

No worries, you've done more than I imagined anyone would do. I appreciate all your help, I imagine at some point I'll start getting busier too with work (hopefully haha)

**Reply by K-dash** [reply]
Oh, really? Are you working as a developer now?

Wow, you were in Japan. What a coincidence!
There are a lot of foreign tourists in Kyoto and Tokyo these days, probably because of the weak yen (haha).

> No worries, you've done more than I imagined anyone would do. I appreciate all your help, I imagine at some point I'll start getting busier too with work (hopefully haha)

I'm glad to hear that. I'll continue to support you as much as my time allows, so let's keep in touch!

**Reply by ryansurf** [reply]
> Oh, really? Are you working as a developer now?

Not yet, I'm looking for work atm. Interested in dev/infrastructure, hopefully I find something soon

Japan was awesome, did all the touristy things but I had a wonderful time. I've heard the yen is weak but it was still pretty expensive haha. The shinkansen was a good experience and the food was delicious :yum::yum::yum:

**Reply by K-dash** [reply]
> Not yet, I'm looking for work atm. Interested in dev/infrastructure, hopefully I find something soon

I hope you find a great job. In Japan, you can become an engineer even without a computer science degree, and it’s rare to get fired, so it's a bit more lenient compared to your country (haha).

By the way, I noticed that your skill set (infrastructure and Python development) is very similar to mine. Let's keep exchanging information!

> apan was awesome, did all the touristy things but I had a wonderful time. I've heard the yen is weak but it was still pretty expensive haha. The shinkansen was a good experience and the food was delicious 😋😋😋

That sounds great! If you have a chance to come to Japan again, I can recommend some good places to visit!

**Reply by ryansurf** [reply]
Thank you! Here in the United States the job market is competitive right now. Not many companies are hiring junior engineers, supply is much higher than demand. That sounds nice over in Japan

Yup! I'm not very skilled with infrastructure yet but its an interest of mine, learning as I go

> That sounds great! If you have a chance to come to Japan again, I can recommend some good places to visit!

Where would you recommend? Out of all the countries I visited in Asia, Japan was my favorite (and I spent the least amount of time there). When I was there I visited Shibuya, did the Philosopher's Walk and went to a night market in Kyoto and did a lot of walking around in Tokyo

**Reply by K-dash** [reply]
> Thank you! Here in the United States the job market is competitive right now. Not many companies are hiring junior engineers, supply is much higher than demand. That sounds nice over in Japan

I see. That sounds tough...
BTW, in Japan, there has been a shortage of engineers relative to the job market demand in recent years, so it has been a seller's market for engineers.

> Where would you recommend? Out of all the countries I visited in Asia, Japan was my favorite (and I spent the least amount of time there). When I was there I visited Shibuya, did the Philosopher's Walk and went to a night market in Kyoto and did a lot of walking around in Tokyo

Nice!
In addition to Tokyo and Kyoto, I also recommend Osaka. There are many delicious foods there.
Also, did you try Japanese ramen? There are numerous ramen shops in Japan, but among them, there are specific types of ramen called "Ie-kei" and "Jiro-kei" which are very delicious. Next time you come to Japan, give them a try! But, They may not be suitable for those who don't like greasy food. haha 🫠

**Comment by K-dash** [comment]
Rest assured. External collaborators can be given detailed permissions, so there is no problem as long as the minimum necessary permissions are granted to them:)

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/permission-levels-for-a-personal-account-repository

**Reply by GUKWAT** [reply]
@ryansurf @K-dash I missed this whole discussion because I got a little bit busy during the week but I agree with @K-dash. As the maintainer you should also submit pull requests for review before merging

**Reply by ryansurf** [reply]
Thanks for the feedback, I'm now submitting PRs!

**Comment by K-dash** [comment]
@ryansurf 
I believe this discussion has also been resolved, so it might be a good idea to mark it as answered and then close it:).
